lib/LaTeXML/resources/CSS/LaTeXML.css

```diff
@@ -436,7 +436,34 @@ span.ltx_framed       { display:inline-block; text-indent:0; } /* avoid padding/
 .ltx_nounicode    { color:red; }
 .ltx_nodisplay    { display:none; }
 /*======================================================================
- SVG (pgf/tikz ?) basics */
+ SVG (pgf/tikz & xy) basics
+ Layout depends heavily on TeX's fonts which are more compact than typical browser's fonts
+ */
+/* Use flex to center the FO's content (horizontally & vertically) over the FO's size */
+
+/* Use width 110% in a couple of places to inhibit unexpected line breaks
+  due to TeX/browser font differences */
+.ltx_foreignobject_container {
+    display:flex;
+    align-items:center;
+    justify-content:center;
+    height:100%;
+    width:110%;
+    position:relative; left:-5%;
+}
+/* line height 0 just for this layer, to keep the line-box compact */
+.ltx_foreignobject_content {
+    display:block;
+    line-height:0;
+}
+/*    line-height:revert;  Why doesn't this work? */
+.ltx_foreignobject_content * {
+    line-height:normal;
```

Comment:
If all of the content is inside a node, I believe this will negate `line-height:0` altogether. Revert is too heavy handed: it will go back to the UA stylesheet which is probably `inherit`. revert-layer has better semantics for this.